linstor/linstor-monitord.c

```diff
@@ -130,24 +133,118 @@ static inline int isMasterHost (int *error) {
 
 typedef struct {
   int inotifyFd;
+  struct timespec lastScanTime;
+  int isMaster;
   // TODO: Should be completed with at least a hostname field.
 } State;
 
 // -----------------------------------------------------------------------------
 
-static inline int execCommand (char *argv[]) {
+typedef struct {
+  char *data;
+  size_t size;
+  size_t capacity;
+} Buffer;
+
+#define max(a, b) ({ \
+  __typeof__(a) _a = (a); \
+  __typeof__(b) _b = (b); \
+  _a > _b ? _a : _b; \
+})
+
+static inline ssize_t readAll (int fd, Buffer *buffer) {
+  assert(buffer->capacity >= buffer->size);
+
+  ssize_t ret = 0;
+  do {
+    size_t byteCount = buffer->capacity - buffer->size;
+    if (byteCount < 16) {
+      const size_t newCapacity = max(buffer->capacity << 1, 64);
+      char *p = realloc(buffer->data, newCapacity);
```

Comment:
Why why?

Reply:
why 16, why 64. Either comment or use named const to be more readable :)

Reply:
It's purely arbitrary and it's common to use a minimum fixed block size to avoid many allocations. :p

Concerning 16: I don't want to execute two "read" calls if there is a big amount of data in the pipe buffer.